pkg/api/clustermanager.go

```diff
@@ -0,0 +1,153 @@
+package api
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+// SyncSetList represents a list of SyncSets for a given cluster.
+type SyncSetList struct {
+	SyncSets []*SyncSet `json:"value"`
+	// The link used to get the next page of operations.
+	NextLink string `json:"nextLink,omitempty"`
+}
+
+type ClusterManagerConfigurationList struct {
+	ClusterManagerConfigurations []*ClusterManagerConfiguration `json:"value"`
+
+	NextLink string `json:"nextLink,omitempty"`
+}
+
+// ClusterManagerConfiguration represents the configuration from OpenShift Cluster Manager (OCM)
+type ClusterManagerConfiguration struct {
+	MissingFields
+
+	// ID is the unique identifier for the cluster manager configuration
+	ID                string                                `json:"id,omitempty"`
+	Name              string                                `json:"name,omitempty"`
+	ClusterResourceID string                                `json:"clusterResourceId,omitempty"`
+	Properties        ClusterManagerConfigurationProperties `json:"properties,omitempty"`
+	SystemData        *SystemData                           `json:"systemData,omitempty"`
+}
+
+// ClusterManagerConfigurationProperties houses the payloads the frontend receives for all OCM resources
+// we store them as a byte slice in cosmos
+type ClusterManagerConfigurationProperties struct {
+	Resources []byte `json:"resources,omitempty"`
+}
+
+// SyncSet represents a SyncSet for an Azure Red Hat OpenShift Cluster.
+type SyncSet struct {
```

Comment:
I'm not a fan of these struct definitions that aren't actually used anywhere since they're essentially boilerplate for ClusterManagerConfiguration.  I understand the usage of them, but maybe in a follow-up PR let's rip it out and refactor frontend.  This change won't affect any swagger definitions so let's leave it as is for now.

Reply:
Technically the incoming request will have a path (which is technically the ID) associated with it so we can generate a switch statement to utilize the underlying resource. 

This is going to tightly couple us with OCM if they release new resources, but we're already there with this so :shrug: